trin-history/src/jsonrpc.rs

```diff
@@ -124,11 +123,15 @@ async fn recursive_find_content(
 
     // If tracing is not required, return content.
     if !is_trace {
-        return Ok(content_response_string);
+        return Ok(json!(ContentInfo::Content {
+            content: serde_json::from_value(content_response_string).map_err(|e| e.to_string())?,
```

Comment:
Something here seems really off. Are we encoding to json string in `content_response_string`, then decoding it to re-encode it? ... I'm seeing now that maybe the same thing is happening in the trace below, but that also seems wrong.

Does this work?
```rust
    // Handle empty content
    let content_string = match possible_content_bytes {
        Some(bytes) => hex_encode(bytes),
        None => CONTENT_ABSENT.to_string(), // "0x"
    };

    // If tracing is not required, return content.
    if !is_trace {
        return Ok(json!(ContentInfo::Content {
            content: content_string,
```